crates/std_detect/src/detect/cache.rs

```diff
@@ -81,83 +77,44 @@ impl Initializer {
 }
 
 /// This global variable is a cache of the features supported by the CPU.
+#[cfg(target_pointer_width = "64")]
 static CACHE: Cache = Cache::uninitialized();
 
-/// Feature cache with capacity for `CACHE_CAPACITY` features.
+/// This global variable is a cache of the features supported by the CPU.
+#[cfg(target_pointer_width = "32")]
+static CACHE: [Cache; 2] = [Cache::uninitialized(), Cache::uninitialized()];
+
+/// Feature cache with capacity for `usize::max_value() - 1` features.
 ///
 /// Note: the last feature bit is used to represent an
 /// uninitialized cache.
-#[cfg(target_pointer_width = "64")]
-struct Cache(AtomicU64);
+struct Cache(AtomicUsize);
 
-#[cfg(target_pointer_width = "64")]
-#[allow(clippy::use_self)]
 impl Cache {
     /// Creates an uninitialized cache.
     #[allow(clippy::declare_interior_mutable_const)]
     const fn uninitialized() -> Self {
-        Cache(AtomicU64::new(u64::max_value()))
+        Cache(AtomicUsize::new(usize::max_value()))
     }
     /// Is the cache uninitialized?
     #[inline]
     pub(crate) fn is_uninitialized(&self) -> bool {
-        self.0.load(Ordering::Relaxed) == u64::max_value()
+        self.0.load(Ordering::Relaxed) == usize::max_value()
     }
 
     /// Is the `bit` in the cache set?
     #[inline]
     pub(crate) fn test(&self, bit: u32) -> bool {
-        test_bit(CACHE.0.load(Ordering::Relaxed), bit)
+        test_bit(self.0.load(Ordering::Relaxed) as u64, bit)
     }
 
     /// Initializes the cache.
     #[inline]
-    pub(crate) fn initialize(&self, value: Initializer) {
-        self.0.store(value.0, Ordering::Relaxed);
+    fn initialize(&self, value: usize) {
+        self.0.store(value, Ordering::Relaxed);
```

Comment:
I would personally say this should all become `SeqCst`. Especially on 32-bit platforms where a write to one location is trying to imply a write to a different location. If it's necessarily discovered later I think we can start relaxing this.

Reply:
The idea behind this PR is exactly to write a code in such a way that each cache instance is independent, so that we are in the single memory location world, where Relaxed works.

I think using `SeqCst` would obscure this intention. Instead, I've added a comment explaining why we use the most reckless memory ordering. 

I can still change this to SeqCst, but I think that would not guarantee anything on top without *also* changing `test` to use `SeqCst`, and `test` looks like a potentially hot method.

Reply:
Ah I think I was mistaken in looking at how the testing was done, I thought when the test for "am I initialized" was done by looking at the first word, not the word we're interested in.

I would still personally prefer to see `SeqCst` everywhere because it hardly matters here in terms of performance and anything that makes atomics "a bit more correct" is something I'm personally a fan of at least.